dotnet/src/SemanticKernel.Abstractions/SkillDefinition/FunctionsView.cs

```diff
@@ -17,19 +18,21 @@ namespace Microsoft.SemanticKernel.SkillDefinition;
 [DebuggerDisplay("{DebuggerDisplay,nq}")]
 public sealed class FunctionsView
 {
+    private object _lock = new();
     /// <summary>
     /// Collection of semantic skill names and function names, including function parameters.
     /// Functions are grouped by skill name.
     /// </summary>
-    public ConcurrentDictionary<string, List<FunctionView>> SemanticFunctions { get; set; }
-        = new(StringComparer.OrdinalIgnoreCase);
+    public IReadOnlyDictionary<string, IReadOnlyCollection<FunctionView>> SemanticFunctions => this.GetFunctions(nativeFunctionCheck: false);
```

Comment:
Is `GetFunctions` method necessary here? 
Can we do just:
```csharp
public IReadOnlyDictionary<string, IReadOnlyCollection<FunctionView>> SemanticFunctions => new ReadOnlyFunctionsView(this._semanticFunctions);
```
Same applies for `NativeFunctions`.